src/actions/websocket.js

```diff
@@ -131,22 +136,26 @@ function doReconnect() {
             }
 
             dispatch(getStatusesByIds(Object.keys(statusesToLoad)));
-            dispatch(fetchMyChannelsAndMembers(currentTeamId)).then(({data}) => {
-                dispatch(loadProfilesForDirect());
-
-                if (data && data.members) {
-                    const stillMemberOfCurrentChannel = data.members.find((m) => m.channel_id === currentChannelId);
-                    if (!stillMemberOfCurrentChannel) {
-                        EventEmitter.emit(General.SWITCH_TO_DEFAULT_CHANNEL, currentTeamId);
-                    }
-                }
-            });
-
-            dispatch(getPosts(currentChannelId));
             dispatch(getMyTeamUnreads());
 
-            const myTeamMembers = getTeamMemberships(getState());
-            if (!myTeamMembers[currentTeamId]) {
+            // We need to wait for these actions so that we have an
+            // up-to-date state of the current user's team memberships.
+            await dispatch(getMyTeams());
+            await dispatch(getMyTeamMembers());
+            const currentTeamMembership = getCurrentTeamMembership(getState());
+            if (currentTeamMembership) {
+                dispatch(getPosts(currentChannelId));
+                dispatch(fetchMyChannelsAndMembers(currentTeamId)).then(({data}) => {
```

Comment:
if you are using async/await then why not changing this from the promise to also use await?

Reply:
I was going to comment the same thing but using async/await here does have a functional change. `dispatch({type: GeneralTypes.WEBSOCKET_SUCCESS});` will now wait for all those requests to complete before being run. Not sure if this is a problem or not but it is a change we should be aware of

Reply:
It actually might be better to wait for the requests to finish since the connection bar will go green once all the updates have been fetched (nothing pops in after the connection bar is green). I'm 0/5 either way